ale_linters/python/pyls.vim

```diff
@@ -0,0 +1,33 @@
+" Author: aurieh <me@aurieh.me>
+" Description: A language server for Python
+
+call ale#Set('python_pyls_executable', 'pyls')
+
+function! ale_linters#python#pyls#GetExecutable(buffer) abort
+    return ale#Var(a:buffer, 'python_pyls_executable')
+endfunction
+
+function! ale_linters#python#pyls#GetLanguage(buffer) abort
+    return 'python'
+endfunction
+
+function! ale_linters#python#pyls#GetProjectRoot(buffer) abort
+    " Start with more generic files
+    for l:possible_filename in ['setup.cfg', 'tox.ini', 'flake8.cfg', 'pycodestyle.cfg']
```

Comment:
I recommend using `ale#python#FindProjectRoot` here, and you could maybe modify that function so it will look for other configuration files. Then the behavior of searching for Python project roots will be consistent everywhere.

Reply:
PyLS only checks for that set of files

Reply:
I could probably use `python#FindProjectRoot`, but it wouldn't be accurate

Reply:
I recommend it. One important difference is that it uses a bread-first search for configuration files instead of depth-first search, so `foo/bar/baz/tox.ini` will be preferred over `foo/setup.cfg`.

Reply:
Yeah, that's a good point. Some people have multiple per-module configs.
I'll just add more filenames to `python#FindProjectRoot`.